File: crates/kndo-adapter-swift/tests/fixtures/visibility-ladder-and-internal-default/project/Sources/App/Widget.swift
open class Widget {
    // Used only within this file — already the tightest rung for its level.
    private func a() {
    }

    // Used only within this file too — `fileprivate` maps to the same `File` scope as
    // `private`, so neither accuses the other even though their ladder indices differ.
    fileprivate func b() {
    }

    // Used from Sibling.swift (same target, different file) — required scope is `Unit`,
    // which the ladder covers at the "internal" rung — already tightest.
    internal func c() {
    }

    // Same evidence as `c`, but declared `public` — wider than the target-only usage
    // requires.
    public func d() {
    }

    // Same evidence as `c`, but declared `open` — wider than the target-only usage
    // requires.
    open func e() {
    }

    // No modifier at all — Swift's default is `internal`. Same target-only evidence as `c`,
    // proving the no-modifier default computes the same tightest-sufficient check as an
    // explicit `internal` would.
    func useOwn() {
    }

    func run() {
        a()
        b()
    }
}

File: v2/crates/kndo-adapter-java/tests/fixtures/visibility-ladder-and-nested-members/project/src/main/java/com/foo/Widget.java
package com.foo;

public class Widget {
    // Used only within this file (main below) — already the tightest rung for its level.
    private void a() {}

    // Used from main() AND Sibling (same package) — required scope is Unit, which matches
    // its OWN declared level exactly — already tightest.
    void b() {}

    // Used ONLY from Sibling (same package, different file) — required is Unit, strictly
    // narrower than protected's declared Public — should flag "package-private would suffice".
    protected void c() {}

    // Same evidence as c(), but declared public — should flag "package-private would suffice".
    public void d() {}

    public static void main(String[] args) {
        Widget w = new Widget();
        w.a();
        w.b();
        Inner in = new Inner();
        in.go();
        new Sibling().run();
    }

    class Inner {
        void go() {
            helper();
        }

        void helper() {}
    }
}
